src/protocol/receiver/time.rs

```diff
@@ -0,0 +1,111 @@
+use std::time::{Duration, Instant};
+
+use stats::OnlineStats;
+
+use crate::protocol::{TimeBase, TimeSpan, TimeStamp};
+
+pub(crate) struct SynchronizedRemoteClock {
+    tolerance: Duration,
+    time_base: TimeBase,
+    stats: Option<OnlineStats>,
+}
+
+impl SynchronizedRemoteClock {
+    const MAX_SAMPLES: usize = 1_000;
+    const DRIFT_TOLERANCE: Duration = Duration::from_millis(5);
+
+    pub fn new(now: Instant) -> Self {
+        Self {
+            // TODO: Drift deviation tolerance should be parameterized.
+            //       It wasn't in the reference implementation, but I added it because the reference
+            //       implementation is susceptible to invalid clock adjustments during periods of
+            //       acute network latency
+            tolerance: Self::DRIFT_TOLERANCE,
+            time_base: TimeBase::new(now),
+            stats: None,
+        }
+    }
+
+    pub fn synchronize(&mut self, now: Instant, ts: TimeStamp) {
+        let drift = self.time_base.timestamp_from(now) - ts;
+        match &mut self.stats {
+            None => {
+                self.time_base.adjust(drift);
+            }
+            Some(stats) => {
+                stats.add(drift);
+
+                if stats.len() < Self::MAX_SAMPLES {
+                    return;
+                }
+
+                if stats.stddev() < self.tolerance.as_micros() as f64 {
```

Comment:
What if it never gets under this threshold?

Reply:
If it never gets under the threshold, then either the network between your hosts or their clocks are erratic, so a sufficiently stable clock sync will not be feasible. This threshold is what is considered sufficiently stable. If clock drift has high deviation, then you have much larger issues. I have no experience using the reference SRT implementation in networks with high congestion or latency variance, yet, so I'm not actually sure this is essential, but I do have experience working with clock sync algorithms in general, and I anticipate this will be an issue. I'm fine with removing this behavior for now.

Reply:
No it makes sense. Accounting for drift seems unreasonable if there is high latency variation, and just keeping with the original clock is likely better than trying to sync.